docker/hail-ubuntu/Dockerfile

```diff
@@ -20,20 +22,25 @@ RUN chmod 755 /bin/retry && \
     chmod 755 /controller.sh && \
     echo "APT::Acquire::Retries \"5\";" > /etc/apt/apt.conf.d/80-retries && \
     mkdir -p /usr/share/keyrings/ && \
-    hail-apt-get-install curl apt-transport-https ca-certificates gnupg jq rsync && \
-    curl 'https://keyserver.ubuntu.com/pks/lookup?search=0xF23C5A6CF475977595C89F51BA6932366A755776&hash=on&exact=on&options=mr&op=get' \
-         | gpg --dearmor > /usr/share/keyrings/deadsnakes-ppa-archive-keyring.gpg && \
-    echo 'deb [signed-by=/usr/share/keyrings/deadsnakes-ppa-archive-keyring.gpg] http://ppa.launchpad.net/deadsnakes/ppa/ubuntu noble main' \
-         >> /etc/apt/sources.list && \
-    echo 'deb-src [signed-by=/usr/share/keyrings/deadsnakes-ppa-archive-keyring.gpg] http://ppa.launchpad.net/deadsnakes/ppa/ubuntu noble main' \
-         >> /etc/apt/sources.list && \
+    hail-apt-get-install \
+        apt-transport-https \
+        ca-certificates \
+        curl \
+        g++ \
+        gcc \
+        gnupg \
+        jq \
+        rsync \
+        && \
     curl 'https://packages.cloud.google.com/apt/doc/apt-key.gpg' \
         | gpg --dearmor -o /usr/share/keyrings/cloud.google.gpg && \
     echo 'deb [signed-by=/usr/share/keyrings/cloud.google.gpg] https://packages.cloud.google.com/apt cloud-sdk main' \
-        >> /etc/apt/sources.list && \
-    hail-apt-get-install python$PYTHON_VERSION-minimal python$PYTHON_VERSION-dev python$PYTHON_VERSION-distutils gcc g++ && \
-    update-alternatives --install /usr/bin/python3 python3 /usr/bin/python$PYTHON_VERSION 1 && \
-    curl https://bootstrap.pypa.io/get-pip.py | python3 && \
-    python3 -m pip install 'pip>=23,<26' && \
+        >> /etc/apt/sources.list
+
+ENV UV_MANAGED_PYTHON=1 UV_PYTHON_INSTALL_DIR=/opt/uv/python VIRTUAL_ENV=/opt/venv
+ENV PATH=$VIRTUAL_ENV/bin:$PATH
+RUN uv python install $PYTHON_VERSION && \
+    uv venv --seed $VIRTUAL_ENV && \
```

Comment:
I'm not sure this is right. `uv python install <version>` makes a python version available, but the `uv venv` line isn't specifying a version, and I'm not sure what version gets requested by default, or if it might decide to use a sytem installed python. I think the most explicit thing is to use `uv venv --python <version>` (in which case `uv python install` may be unnecessary).

Reply:
Read through the docs again and found I could just set `UV_PYTHON` and everything will just work. Thanks!